bigbone/src/main/kotlin/social/bigbone/api/entity/WebPushSubscription.kt

```diff
@@ -0,0 +1,94 @@
+package social.bigbone.api.entity
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+/**
+ * Represents a subscription to the push streaming server.
+ * @see <a href="https://docs.joinmastodon.org/entities/WebPushSubscription/">Mastodon API Push</a>
+ */
+@Serializable
+data class WebPushSubscription(
+
+    /**
+     * The ID of the Web Push subscription in the database.
+     */
+    @SerialName("id")
+    val id: String,
+
+    /**
+     * Where push alerts will be sent to.
+     */
+    @SerialName("endpoint")
+    val endpoint: String,
+
+    /**
+     * The streaming server’s VAPID key.
+     */
+    @SerialName("server_key")
+    val serverKey: String,
+
+    /**
+     * Which alerts should be delivered to the endpoint.
+     */
+    @SerialName("alerts")
+    val alerts: Alerts
```

Comment:
Was there a reason why you added default values to all `Alerts` properties, but not to these vals up here?

Reply:
I do not think you need to pre-initialize the nested object, am I wrong? I decided to add default boolean values to every attribute of Alerts just like it is written within the documentation. By the way, would it better if I put them not-nullable? They are not reported like required but I do not if it is useless. https://docs.joinmastodon.org/entities/WebPushSubscription/

Reply:
> I do not think you need to pre-initialize the nested object, am I wrong?

I meant all of them—`id`, `endpoint`, `serverKey`, and `alerts`—, not just `alerts`. But I’m not so sure myself. If they’re _always_ in the payload of the JSON response, there should not be any issue. If they aren’t, we would get `null` values which are not allowed on those non-nullable types.

> By the way, would it better if I put them not-nullable? They are not reported like required but I do not if it is useless

I never really know either… I’ve seen cases in the documentation where parameters were not marked as “required” but also not as “optional”, and then they were returned as some type in one example but then as `null` in another example. Hard to say, really.

As even in the example, not all of the possible properties are available, I assume it could make sense to have them as nullable. But I’m not sure.